### ISSUE TITLE
gRPC AuthService HeaderOptionValues with append=true only work if header already exists

### ISSUE THREAD

**Issue by LukeShu** [datawire/ambassador#1313]
**Describe the bug**

Setting `append=true` in a HeaderValueOption in a gRPC AuthService response causes that HeaderValueOption to do nothing if a header with that name doesn't already exist.

**To Reproduce**

Deploy this trivial Go gRPC AuthService:
```go
package main

import (
	"context"
	"log"
	"net"

	"github.com/gogo/googleapis/google/rpc"
	"github.com/gogo/protobuf/types"
	"google.golang.org/grpc"

	envoyCoreV2 "github.com/datawire/kat-backend/xds/envoy/api/v2/core"
	envoyAuthV2 "github.com/datawire/kat-backend/xds/envoy/service/auth/v2alpha"
)

func main() {
	socket, err := net.Listen("tcp", ":3000") // #nosec G102
	if err != nil {
		log.Fatal(err)
	}

	grpcServer := grpc.NewServer()
	envoyAuthV2.RegisterAuthorizationServer(grpcServer, &AuthService{})

	log.Print("starting...")
	log.Fatal(grpcServer.Serve(socket))
}

type AuthService struct{}

func (s *AuthService) Check(ctx context.Context, req *envoyAuthV2.CheckRequest) (*envoyAuthV2.CheckResponse, error) {
	return &envoyAuthV2.CheckResponse{
		Status: &rpc.Status{Code: int32(rpc.OK)},
		HttpResponse: &envoyAuthV2.CheckResponse_OkResponse{
			OkResponse: &envoyAuthV2.OkHttpResponse{
				Headers: []*envoyCoreV2.HeaderValueOption{
					{Header: &envoyCoreV2.HeaderValue{Key: "X-Override", Value: "foo"}, Append: &types.BoolValue{Value: false}},
					{Header: &envoyCoreV2.HeaderValue{Key: "X-Append", Value: "bar"}, Append: &types.BoolValue{Value: true}},
				},
			},
		},
	}, nil
}
```
Using this YAML:
```yaml
# @TEMPLATE@

---
apiVersion: v1
kind: Service
metadata:
  name: grpcauthservice
  annotations:
    getambassador.io/config: |
      ---
      apiVersion: ambassador/v1
      kind: AuthService
      name: authentication
      proto: grpc
      auth_service: grpcauthservice
spec:
  type: ClusterIP
  selector:
    service: grpcauthservice
  ports:
  - name: auth-grpc
    port: 80
    targetPort: 3000
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: grpcauthservice
  labels:
    service: grpcauthservice
spec:
  replicas: 1
  selector:
    matchLabels:
      service: grpcauthservice
  template:
    metadata:
      labels:
        service: grpcauthservice
    spec:
      containers:
        - name: grpcauthservice
          image: {{env "GRPCAUTHSERVICE_IMAGE"}}
          ports:
            - name: auth-grpc
              containerPort: 3000

---
apiVersion: v1
kind: Service
metadata:
  name: httpbin
  annotations:
    getambassador.io/config: |
      ---
      apiVersion: ambassador/v0
      kind: Mapping
      name: httpbin_mapping
      prefix: /
      service: httpbin.org:80
      host_rewrite: httpbin.org
      timeout_ms: 5000
spec:
  type: ClusterIP
  ports:
  - name: httpbin
    port: 80
```
Run  this in a terminal:
```
$ curl http://ambassador.default.svc.cluster.local/headers
{
  "headers": {
    "Accept": "*/*", 
    "Host": "httpbin.org", 
    "User-Agent": "curl/7.64.0", 
    "X-Envoy-Expected-Rq-Timeout-Ms": "5000", 
    "X-Envoy-Internal": "true", 
    "X-Envoy-Original-Path": "/headers", 
    "X-Override": "foo"
  }
}

$ curl -H 'X-Override: before' -H 'X-Append: before' http://ambassador.default.svc.cluster.local/headers
{
  "headers": {
    "Accept": "*/*", 
    "Host": "httpbin.org", 
    "User-Agent": "curl/7.64.0", 
    "X-Append": "before,bar", 
    "X-Envoy-Expected-Rq-Timeout-Ms": "5000", 
    "X-Envoy-Internal": "true", 
    "X-Envoy-Original-Path": "/headers", 
    "X-Override": "foo"
  }
}
```

**Expected behavior**

I expected the first `curl` command to also include a line saying
```
    "X-Append": "bar", 
```
Instead, it had no `X-Append` line.

**Versions (please complete the following information):**
 - Ambassador: 0.50.2, 0.51.2
 - Kubernetes environment: Kubernaut
 - Version: 1.10.5

**Comment by gsagula** [comment]
This could be a problem with our Envoy fork. I will look into that shortly.

**Comment by kflynn** [comment]
Fixed in 0.52.0.
